akka-sample-sharding-scala/killrweather/src/main/scala/sample/killrweather/JsonFormats.scala

```diff
@@ -9,15 +13,39 @@ object JsonFormats {
   import spray.json.RootJsonFormat
   // import the default encoders for primitive types (Int, String, Lists etc)
   import spray.json.DefaultJsonProtocol._
+  import spray.json.deserializationError
+
+  /**
+   * Given a set of possible case objects, create a format that accepts uses their toString representation in json
+   */
+  private final case class SimpleEnumFormat[T](possibleValues: Set[T]) extends JsonFormat[T] {
+    val stringToValue: Map[String, T] = possibleValues.map(value => value.toString.toLowerCase -> value).toMap
+    val valueToString: Map[T, String] = stringToValue.map { case (string, value) => value -> string }
+
+    override def read(json: JsValue): T = json match {
+      case JsString(text) =>
+        stringToValue.get(text.toLowerCase) match {
+          case Some(t) => t
+          case None => deserializationError(s"Possible values are ${stringToValue.keySet}, [$text] is not among them")
+        }
+      case surprise =>
+        deserializationError(s"Expected a string value, got $surprise")
+    }
+
+    override def write(obj: T): JsValue =
+      JsString(valueToString(obj))
+  }
```

Comment:
Before the json representation of the function and datatype was pushed to the actual types (but then never used), I think it rather belongs here, even if a bit more verbose